### ISSUE TITLE
Windows 10 install via pip

### ISSUE THREAD

**Issue by ydkhatri** [netleibi/fastchunking#6]
It seems installing via pip fails, while installing manually `python setup.py build` and `python setup.py install` works. Can you look into this?

This is using python 3.9.6 on Windows 10 21H1 (19043)

Error when using pip:
```
C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\bin\HostX86\x64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:c:\python39\libs /LIBPATH:c:\python39\PCbuild\amd64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\ATLMFC\lib\x64 /LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.29.30037\lib\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\lib\um\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\ucrt\x64 /LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.19041.0\um\x64 /EXPORT:PyInit__rabinkarprh build\temp.win-amd64-3.9\Release\Users\superman\AppData\Local\Temp\pip-install-u3ks__1d\fastchunking_d3faaf24a59444849d58f77d870f172e\build\rabinkarprh.obj build\temp.win-amd64-3.9\Release\lib/rabinkarp.obj /OUT:build\lib.win-amd64-3.9\fastchunking\_rabinkarprh.cp39-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.9\Release\Users\superman\AppData\Local\Temp\pip-install-u3ks__1d\fastchunking_d3faaf24a59444849d58f77d870f172e\build\_rabinkarprh.cp39-win_amd64.lib
    LINK : fatal error LNK1104: cannot open file 'build\temp.win-amd64-3.9\Release\Users\superman\AppData\Local\Temp\pip-install-u3ks__1d\fastchunking_d3faaf24a59444849d58f77d870f172e\build\_rabinkarprh.cp39-win_amd64.exp'
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\BuildTools\\VC\\Tools\\MSVC\\14.29.30037\\bin\\HostX86\\x64\\link.exe' failed with exit code 1104
```

**Comment by netleibi** [comment]
Thanks for reporting!

I could reproduce the problem with Python 3.9 on Windows 10 and I've just spent way too much time investigating this issue. As it turns out, the problem is neither specific to fastchunking nor to Python 3.9. Instead, it has been caused by an upgrade of pip: Installation seems to work flawlessly until pip 20.2.4, but it fails with pip >= 20.3.

With that version, pip seems to have changed its naming convention for the temporary directory that is used for building the package, leading to longer paths.

Before: `C:\Users\leibi\AppData\Local\Temp\pip-install-tgv2oi69\fastchunking`
After: `C:\Users\leibi\AppData\Local\Temp\pip-install-tgv2oi69\fastchunking_19d8102fa81b4e35b8ce2e30f36bc467`

The build of the C++ extension used by fastchunking happens in a subdirectory that reflects the whole file path, i.e., is performed in a subdirectory such as:

`C:\Users\leibi\AppData\Local\Temp\pip-install-tgv2oi69\fastchunking_19d8102fa81b4e35b8ce2e30f36bc467\build\temp.win-amd64-3.9\Release\Users\leibi\AppData\Local\Temp\pip-install-tgv2oi69\fastchunking_19d8102fa81b4e35b8ce2e30f36bc467\build`

The linker is supposed to create a file named `_rabinkarprh.cp39-win_amd64.exp` under that directory, as mentioned in your error message. The resulting path, though, exceeds Windows' 260-characters file path limitation (see https://docs.microsoft.com/windows/win32/fileio/maximum-file-path-limitation), causing the process to fail.

Unfortunately, enabling "Win32 long paths" support as described under the link above does not resolve the problem, as long paths would have to be explicitly supported by the Visual C++ linker, which is apparently not the case.

The best solution I have come up so far is to change the temporary directory used by `pip install`. pip seems to respect the `TMPDIR` environment variable, so the following command sequence works when a directory `C:\temp` exists:

```
set TMPDIR=C:\temp
pip install fastchunking
```

I am open for suggestions for a suitable workaround that can be implemented inside this library, though.
